core/src/main/scala/org/apache/spark/deploy/ExecutorDescription.scala

```diff
@@ -25,10 +25,13 @@ package org.apache.spark.deploy
 private[deploy] class ExecutorDescription(
     val appId: String,
     val execId: Int,
+    val rpId: Int,
     val cores: Int,
+    val memoryMb: Int,
```

Comment:
Why do we need `memoryMb` now?

Reply:
> And in master, we can only reconstruct the resource profile information in ApplicationInfo after client send resource request RequestExecutors

Does it mean we can't launch new executors with the specific `rpId` until the client sends the request with the corresponding resource profile? For example, if the number of executos with the specific `rpId` hasn't reached the targer number, it seems like we can't schedule new executors for it until we know resource profile by `RequestExecutors`, right?

Reply:
Yes, correct. Master won't allocate new executors before next resource requests.